### DISCUSSION TITLE
Vertical redesign of the product addition form

### DISCUSSION THREAD

**Discussion by teolemon** [openfoodfacts/openfoodfacts-androidapp#3961] [Ideas]
![image](https://user-images.githubusercontent.com/1689815/52914851-9a4e0680-32cd-11e9-8e80-2362f5490d38.png)

- Uploading the 3 pictures is the 1st thing the user has to do. They are immediately sent to the server.
- Processing of the ingredients is started as soon as the ingredient picture is uploaded
- Explanation that the category is important for the Nutri-Score
- Save button are regularly placed within the form to make it clear that the user can add as much, or as little info as she/he wants
- Using the minimal mode (not all fields are displayed, just the ones required to Compute the Nutri-Score and Nova)

**Comment by subhanjansaha** [comment]
Working on this!! :-)

**Comment by subhanjansaha** [comment]
I love the improvements suggested. They would simultaneously streamline the process of adding a product and improve the quality of contributions.

**Comment by teolemon** [comment]
Also usability improvement: the name of the nutrient should be visible, even after a value has been typed.
![image](https://user-images.githubusercontent.com/1689815/52914997-0d0bb180-32cf-11e9-9ab6-315bd38f8170.png)

**Comment by teolemon** [comment]
@subhanjansaha We tried the vertical approach for iOS, and it seems less intimidating that the current solution, shared with the edit mode.

**Comment by teolemon** [comment]
![image](https://user-images.githubusercontent.com/1689815/52915088-121d3080-32d0-11e9-96b2-5c82e8905470.png)

**Comment by subhanjansaha** [comment]
That's very true. Very often, when editing a product, it's extremely confusing to figure which value represents which nutrient for a product. 

See attached,
![screenshot_20190217-204656](https://user-images.githubusercontent.com/42271776/52915083-bbc2e880-32f5-11e9-8e57-30a9226da8e2.png)

**Comment by subhanjansaha** [comment]
@teolemon I completely agree on every aspect. As a user of the app myself, I can vouch for the fact that when adding a new product, having to fill in a whole new form for a product seems like a tedious task to do. Instead, this redesign would keep the 'intimidation' to a minimum and allow an effortless entry of data.
